_posts/2021-08-30-release-statefun-3.1.0.md

```diff
@@ -0,0 +1,228 @@
+and the [updated documentation](https://ci.apache.org/projects/flink/flink-statefun-docs-release-3.0/).
+We encourage you to download the release and share your feedback with the community through the [Flink mailing lists](https://flink.apache.org/community.html#mailing-lists)
+or [JIRA](https://issues.apache.org/jira/browse/)!
+
+{% toc %}
+
+## New Features 
+
+### Delayed Message Cancellation
+
+Stateful Functions communicate by sending messages, but sometimes it is helpful that a function will send a message for itself.
+For example, set a limit for a customer onboarding flow to complete, for that StateFun support sending a message after a certain delay had passed.
+But up until now, there was no way to indicate to the StateFun runtime that a particular delayed message is not necessary anymore (a customer had completed their onboarding flow) 
+With StateFun 3.1, it is now possible to cancel a delayed message.
+
+```python
+...
+context.send_after(timedelta(days=3),
+                  message_builder(target_typename="fns/onboarding",
+                                  target_id="user-1234",
+                                  str_value="send a reminder email"),
+                  cancellation_token="flow-1234")
+...
+```
+
+To cancel the message at a later time, simply call 
+
+```python
+context.cancel_delayed_message("flow-1234")
+```
+
+Please note that a message cancellation occurs on a best-effort basis, as the message might already be delivered or enqueued for immediate delivery on a remote worker’s mailbox.
+
+### New way to specify components 
+
+StateFun applications consist of multiple configuration components, including remote function endpoints, along with ingress and egress definitions, defined in a YAML format. 
+We've added a new structure that treats each StateFun component as a standalone YAML document in this release.
+Thus, a `module.yaml` file becomes simply a collection of components.
+
+```yaml
+kind: io.statefun.endpoints.v2/http
+spec:
+  functions: com.example/*
+  urlPathTemplate: https://bar.foo.com/{function.name}
+---
+kind: io.statefun.kafka.v1/ingress
+spec:
+  id: com.example/my-ingress
+  address: kafka-broker:9092
+  consumerGroupId: my-consumer-group
+  topics:
+    - topic: message-topic
+      valueType: io.statefun.types/string
+      targets:
+        - com.example/greeter
+---
+kind: io.statefun.kafka.v1/egress
+spec:
+  id: com.example/my-egress
+  address: kafka-broker:9092
+  deliverySemantic:
+    type: exactly-once
+    transactionTimeout: 15min
+---
+```
+
+While this might seem like a minor cosmetic improvement, this change opens the door to more flexible configuration management options in future releases - such as managing each component as a custom K8s resource definition or even behind a REST API. StateFun still supports the legacy module.yaml file for backward compatibility, but users are encouraged to upgrade. 
```

Comment:
```suggestion
While this might seem like a minor cosmetic improvement, this change opens the door to more flexible configuration management options in future releases - such as managing each component as a custom K8s resource definition or even behind a REST API. StateFun still supports the legacy module format in version 3.0 for backward compatibility, but users are encouraged to upgrade. 
```